Lambdas/GHOAuth/OAuthLambda.swift

```diff
@@ -0,0 +1,124 @@
+            return try await handleThrowing(event, context: context)
+        } catch {
+            do {
+                try await discordClient.createMessage(
+                    channelId: OAuthLambdaConstants.Channels.logs.id,
+                    payload: .init(embeds: [.init(
+                        title: "GHOAuth lambda top-level error",
+                        description: String("\(error)".prefix(4_000)),
+                        color: .red
+                    )])
+                ).guardSuccess()
+            } catch {
+                logger.error("DiscordClient logging error", metadata: [
+                    "error": "\(error)"
+                ])
+            }
+            throw error
+        }
+    }
+
+    func handleThrowing(
+        _ event: APIGatewayV2Request,
+        context: LambdaContext
+    ) async throws -> APIGatewayV2Response {
+        logger.trace("Received event: \(event)")
+
+        guard let code = event.queryStringParameters?["code"] else {
+            return .init(
+                statusCode: .badRequest,
+                body: "Missing code query parameter"
+            )
+        }
+
+        logger.trace("Retrieving secrets")
+        let clientSecret = try await secretsRetriever.getSecret(arnEnvVarKey: "GH_CLIENT_SECRET_ARN")
+        guard let clientID = ProcessInfo.processInfo.environment["GH_CLIENT_ID"] else {
+            throw OAuthLambdaError.envVarNotFound(name: "GH_CLIENT_ID")
+        }
+
+        // https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/authorizing-oauth-apps
+
+        logger.trace("Requesting GitHub access token")
+        var request = HTTPClientRequest(url: "https://github.com/login/oauth/access_token")
+        request.method = .POST
+        request.headers = [
+            "Accept": "application/json",
+            "Content-Type": "application/json"
+        ]
+        let requestBody = try JSONEncoder().encode([
+            "client_id": clientID,
+            "client_secret": clientSecret,
+            "code": code
+        ])
+        request.body = .bytes(requestBody)
+
+        let response = try await client.execute(request, timeout: .seconds(30))
+        logger.trace("Got response: \(response.status)")
+
+        guard response.status == .ok else {
+            throw OAuthLambdaError.unexpectedResponse(statusCode: Int(response.status.code))
+        }
+
+        let responseBody = try await response.body.collect(upTo: 1024 * 1024)
+        let accessToken = try JSONDecoder().decode(AccessTokenResponse.self, from: responseBody).accessToken
+
+        // https://docs.github.com/en/rest/users/users?apiVersion=2022-11-28#get-the-authenticated-user
+
+        logger.trace("Requesting GitHub user info")
+
+        request = HTTPClientRequest(url: "https://api.github.com/user")
+        request.method = .GET
+        request.headers = [
+            "Accept": "application/vnd.github+json",
+            "Authorization": "Bearer \(accessToken)"
```

Comment:
i guess you can add the `X-GitHub-Api-Version: 2022-11-28` header too like Github says, just to make sure they don't break our code.